test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetVerifyTests
+    {
+        private readonly string _noTimestamperWarningCode = NuGetLogCode.NU3027.ToString();
+        private readonly string _primarySignatureInvalidErrorCode = NuGetLogCode.NU3018.ToString();
+        private readonly string _noMatchingCertErrorCode = NuGetLogCode.NU3034.ToString();
+        private readonly string _notSignedErrorCode = NuGetLogCode.NU3004.ToString();
+
+        private MsbuildIntegrationTestFixture _msbuildFixture;
+
+        public DotnetVerifyTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _msbuildFixture = fixture;
+        }
+
+        [CIOnlyFact]
+        public async Task Verify_UnSignedPackage_Fails()
+        {
+            using (var packageDir = TestDirectory.Create())
+            {
+                var packageId = "Unsigned.PackageX";
+                var packageVersion = "1.0.0";
+                var packageFile = await TestPackagesCore.GetRuntimePackageAsync(packageDir, packageId, packageVersion);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    packageDir,
+                    $"nuget verify {packageFile.FullName}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                result.Output.Should().Contain(_notSignedErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--all")]
+        [InlineData("")]
```

Comment:
I will change the test inputs. I have created another [isue ](https://github.com/NuGet/Home/issues/10028) because `nuget verify -All` and `nuget verify -Signatures` option has no differences in [implementation](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyCommandRunner.cs#L38).

Reply:
Pasting my comment from there: 

Pretty sure this is by design.

We wanted to allow ourselves the flexibility of adding more validations that are run which people run verify.

The command is not supposed to be constrained to signature verification only.

The idea was that `--all` might run additional validations such as package quality checks. We just haven't gotten to that yet.

Reply:
fyi @JonDouglas 

Just providing context on certain decisions in the past, and I know that you are hoping to get to package quality checks sooner rather than later.

Reply:
Thanks for the inputs @nkolev92. 

> Are `` and --all supposed to be equivalent?

The behavior is equivalent for these two inputs. As per [spec](https://github.com/NuGet/Home/blob/dev/designs/DotNetSignCommands.md#dotnet-nuget-verify), `--all option specifies that all verifications possible should be performed on the package(s). By default, only signatures are verified. ` 

I wanted to test that the `verify` command accepts `--all` switch and verifies signatures for both inputs. Thoughts?

Reply:
If a new verification is added in the future, then no arguments and `--all` will have different behaviour. This test will fail that validation. I don't know if you want to split it into two tests now, or make it the responsibility of the person who adds a new verification (who will also need to understand the context of why this test expects them to have the same behaviour).  In other words, it's best when tests assert business rules, rather than implementation.

Reply:
> If a new verification is added in the future, then no arguments and --all will have different behavior. This test will fail that validation.

By default, only signatures are verified. If a new verification is added in the future both `no arguments` and `--all` should still, verify signature is my understanding. Please let me know if I am missing anything here.

Reply:
I have Just pushed a commit where option `--all` is [hardcoded](https://github.com/NuGet/NuGet.Client/blob/dev-kartheep-ms-dotnet-nuget-verify/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs#L50) in the test itself instead of accepting it as a parameter. The reason for this change is, there are other integration tests where input parameters don't have `--all` switch which implies that both the scenarios are tested in different tests. Hopefully, the new tests don't increase our CI time.